solar_system.rb

```diff
@@ -0,0 +1,52 @@
+require_relative "planet"
+
+class SolarSystem
+  #helper method to create readable but not re-writable instance variable. specified using symbols
+  attr_reader :star_name, :planets
+
+  # constructor method
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  def list_planets
+    n = 0
+    listing = @planets.each do |planet|
+      "#{n += 1}. #{planet.name}"
+    end
+    return listing
+  end
+
+  def user_adds_planet
+    puts "Please enter the details of your new planet: "
+    print "Name: "
+    new_name = gets.chomp.downcase
+    print "Color: "
+    new_color = gets.chomp
+    print "Mass in kg: "
+    mass = gets.chomp
+    print "Distance from #{star_name}: "
+    distance = gets.chomp
+    print "Cool facts about your planet: "
+    fun_deets = gets.chomp
+    new_planet = Planet.new(name: new_name, color: new_color, mass_kg: mass, distance_from_sun_km: distance, fun_fact: fun_deets)
+    add_planet(new_planet)
```

Comment:
This information gathering, seeing as it is coming from the user and talking to the user, generally makes more sense as a part of main to me.